Sources/Jobs/MemoryJobQueue.swift

```diff
@@ -72,6 +72,7 @@ public final class MemoryQueue: JobQueueDriver {
     fileprivate actor Internal {
         var queue: Deque<QueuedJob<JobID>>
         var pendingJobs: [JobID: ByteBuffer]
+        var delayedJobs: [JobID: Date]
```

Comment:
Would it not be cleaner to include the delay date in the queue instead of a separate delayedJobs dictionary. 
```
var queue: Deque<(job: QueuedJob<JobID>, delayedUntil: Date?)>
```
This would mean when you pop a job off the queue that is not ready you are only editing the queue when you place it back on the queue.

Reply:
That's much cleaner indeed!